-fexceptions:
ld: error: undefined symbol: _Unwind_Resume

Should match b_lundef in meson(1).

Skip -Wl,-no_warn_duplicate_libraries on octeon/mips64 to unbreak build:
/usr/bin/ld: attempted static link of dynamic object `/usr/local/lib/libglib-2.0.so.4201.15'

Skip -Wl,--version-script= on octeon/mips64 to unbreak fake:
RuntimeError: Tried to read past the end of the file
    This is a Meson bug and should be reported!

Index: meson.build
--- meson.build.orig
+++ meson.build
@@ -270,7 +270,6 @@ array_bounds = get_option('b_sanitize') == 'none' ? 2 
 
 cc_flags += [
   '-fasynchronous-unwind-tables',
-  '-fexceptions',
   '-fipa-pure-const',
   '-fno-common',
   '-Wabsolute-value',
@@ -560,11 +559,6 @@ libvirt_nodelete = cc.get_supported_link_arguments([
 ])
 
 libvirt_no_undefined = []
-if get_option('b_sanitize') == 'none'
-  libvirt_no_undefined += cc.get_supported_link_arguments([
-    '-Wl,-z,defs',
-  ])
-endif
 
 libvirt_no_indirect = cc.get_supported_link_arguments([
   '-Wl,--no-copy-dt-needed-entries',
@@ -573,10 +567,13 @@ libvirt_no_indirect = cc.get_supported_link_arguments(
 libvirt_no_warn_duplicate_libraries = cc.get_supported_link_arguments([
   '-Wl,-no_warn_duplicate_libraries',
 ])
+if host_machine.system() == 'openbsd' and host_machine.cpu_family() == 'mips64'
+libvirt_no_warn_duplicate_libraries = ''
+endif
 
 if host_machine.system() == 'windows'
   version_script_flags = '-Wl,'
-elif host_machine.system() == 'darwin'
+elif host_machine.system() == 'darwin' or (host_machine.system() == 'openbsd' and host_machine.cpu_family() == 'mips64')
   # macOS libraries don't support symbol versioning
   version_script_flags = ''
 else
@@ -872,7 +869,7 @@ if not get_option('driver_remote').disabled()
     xdr_dep = cc.find_library('portablexdr', required: get_option('driver_remote'))
   elif host_machine.system() in [ 'linux', 'gnu' ]
     xdr_dep = dependency('libtirpc', required: get_option('driver_remote'))
-  elif host_machine.system() in [ 'freebsd', 'darwin' ]
+  elif host_machine.system() in [ 'freebsd', 'darwin', 'openbsd' ]
     xdr_dep = cc.find_library('c', required: get_option('driver_remote'))
   else
     xdr_dep = dependency('', required: false)
@@ -960,15 +957,18 @@ if devmapper_dep.found()
   conf.set('WITH_DEVMAPPER', 1)
 endif
 
-dlopen_use = host_machine.system() != 'windows'
-dlopen_dep = cc.find_library('dl', required: dlopen_use)
-if dlopen_dep.found()
-  if not cc.check_header('dlfcn.h')
-    error('Unable to find dlfcn.h')
-  endif
-  conf.set('WITH_DLFCN_H', 1)
-endif
+#dlopen_use = host_machine.system() != 'windows'
+#dlopen_dep = cc.find_library('dl', required: dlopen_use)
+#if dlopen_dep.found()
+#  if not cc.check_header('dlfcn.h')
+#    error('Unable to find dlfcn.h')
+#  endif
+#  conf.set('WITH_DLFCN_H', 1)
+#endif
 
+dlopen_dep = []
+conf.set('WITH_DLFCN_H', 1)
+
 fuse_version = '3.1.0'
 fuse_dep = dependency('fuse3', version: '>=' + fuse_version, required: false)
 if fuse_dep.found()
@@ -1906,7 +1906,7 @@ endif
 
 driver_modules_flags = []
 if conf.has('WITH_LIBVIRTD')
-  if not conf.has('WITH_DLFCN_H') or not dlopen_dep.found()
+  if not conf.has('WITH_DLFCN_H') #or not dlopen_dep.found()
     error('You must have dlfcn.h / dlopen() support to build driver modules')
   endif
   driver_modules_flags = libvirt_export_dynamic
